test/plain.jl

```diff
@@ -332,4 +332,12 @@ using Compat.String
         str = read(fid["test"])
         @test str == "Hello World"
     end
+    
+    # Test Float16 support
+    arr_float16 = Float16[1.0, 0.25, 0.5, 8.0]
+    h5write("test_float16.h5", "x", arr_float16)
```

Comment:
I guess if you want it to stick around afterwards for debugging, maybe? But then it sticks around in the package dir (I think that's where Pkg.test will usually run from). It's not checked in so it wouldn't make the package dirty in the git sense, but it would be an untracked file.

Would be consistent with the earlier tests to put it under `joinpath(tmpdir, "test_float16.h5")` and move the `rm(tmpdir, recursive=true)` to after this test.